<div style="background:linear-gradient(135deg,#1e1b4b 0%,#4338ca 55%,#6366f1 100%);border-radius:18px;padding:34px 30px;color:#fff;font-family:Inter,Segoe UI,sans-serif">
  <div style="font-size:12px;letter-spacing:3px;color:#c7d2fe;font-weight:700;text-transform:uppercase">Chapter 33 · Solutions</div>
  <div style="font-size:34px;font-weight:900;line-height:1.1;margin:10px 0 6px">Conditional Probability, Worked Answers ✅</div>
  <div style="font-size:15px;color:#e0e7ff;max-width:720px;line-height:1.6">Full solutions to the five challenges. Try them yourself first, then compare.</div>
  <div style="margin-top:16px;font-size:13px;color:#cbd5e1">Statistics, Data Science and AI: A Visual Handbook · John Fisher · 2026</div>
</div>

## ⚙️ Setup

In [1]:
import numpy as np
rng = np.random.default_rng(331)
print("ready")

ready


<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#4f46e5;letter-spacing:1px">CHALLENGE 1 · FROM A TABLE</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Read P(A | B) off the counts</div>
<div style="color:#4a5578;margin-top:6px">Of 200 students, 120 study Python; 90 study Python AND statistics; 130 study statistics. Find P(statistics | Python).</div>
</div>

In [2]:
py, py_and_stat, stat, total = 120, 90, 130, 200
p_stat_given_py = py_and_stat / py    # restrict to the 120 Python students
print(f"P(statistics | Python) = 90/120 = {p_stat_given_py:.3f}")
print(f"compare P(statistics) = 130/200 = {stat/total:.3f}  -> Python students are more likely to take stats (dependent)")

P(statistics | Python) = 90/120 = 0.750
compare P(statistics) = 130/200 = 0.650  -> Python students are more likely to take stats (dependent)


**Answer:** condition on Python, so the denominator is the 120 Python students, not all 200: P(statistics | Python) = 90/120 = 0.75. That is higher than the overall P(statistics) = 0.65, so the two subjects are not independent, studying Python is associated with studying statistics.

<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#2563eb;letter-spacing:1px">CHALLENGE 2 · WITHOUT REPLACEMENT</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">The multiplication rule</div>
<div style="color:#4a5578;margin-top:6px">A bag has 5 red and 3 blue marbles. You draw 2 without replacement. Find P(both red), and verify by simulation.</div>
</div>

In [3]:
p = (5/8) * (4/7)
print(f"P(both red) = (5/8) x (4/7) = {p:.4f}")
bag = np.array([1]*5 + [0]*3)
N=300_000; hits=sum(rng.choice(bag,2,replace=False).sum()==2 for _ in range(N))
print(f"empirical (300k) = {hits/N:.4f}")

P(both red) = (5/8) x (4/7) = 0.3571


empirical (300k) = 0.3566


**Answer:** the draws are dependent, so use the multiplication rule with a conditional second term: P(both red) = P(1st red) x P(2nd red | 1st red) = (5/8) x (4/7) = 20/56 ≈ 0.357. Removing the first red marble changes the bag from 5/8 red to 4/7 red.

<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#059669;letter-spacing:1px">CHALLENGE 3 · INDEPENDENT?</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Test with conditioning</div>
<div style="color:#4a5578;margin-top:6px">A card is drawn from 52. Let A = {it is a King}, B = {it is a Heart}. Is A independent of B? Check whether P(A | B) = P(A).</div>
</div>

In [4]:
pA = 4/52                     # four kings
pA_given_B = 1/13             # of the 13 hearts, one is a king
print(f"P(King)          = 4/52 = {pA:.4f}")
print(f"P(King | Heart)  = 1/13 = {pA_given_B:.4f}")
print("equal -> A and B are INDEPENDENT" if np.isclose(pA,pA_given_B) else "not independent")

P(King)          = 4/52 = 0.0769
P(King | Heart)  = 1/13 = 0.0769
equal -> A and B are INDEPENDENT


**Answer:** P(King) = 4/52 = 1/13, and P(King | Heart) = 1/13 (one king among the 13 hearts). They are equal, so being a King is independent of being a Heart: knowing the suit tells you nothing about whether it is a king. (Suit and rank are independent in a standard deck.)

<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#db2777;letter-spacing:1px">CHALLENGE 4 · BASE RATE</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">The screening trap</div>
<div style="color:#4a5578;margin-top:6px">A disease has 2% prevalence. A test has 95% sensitivity and 90% specificity. In a town of 10,000, find P(disease | positive).</div>
</div>

In [5]:
N=10_000; prev=0.02; sens=0.95; spec=0.90
sick=N*prev; healthy=N*(1-prev)
TP=sick*sens; FP=healthy*(1-spec)
ppv=TP/(TP+FP)
print(f"sick={sick:.0f}, healthy={healthy:.0f}")
print(f"true positives = {TP:.0f}, false positives = {FP:.0f}")
print(f"P(disease | positive) = {TP:.0f}/({TP:.0f}+{FP:.0f}) = {ppv:.3f}  (about {ppv*100:.0f}%)")

sick=200, healthy=9800
true positives = 190, false positives = 980
P(disease | positive) = 190/(190+980) = 0.162  (about 16%)


**Answer:** 200 are sick and 9,800 healthy. True positives = 200 x 0.95 = 190; false positives = 9,800 x 0.10 = 980. So P(disease | positive) = 190 / (190 + 980) ≈ 0.16, only about 16%, because the 10% false-positive rate applied to the large healthy group swamps the true positives. Specificity matters enormously when the disease is rare.

<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#7c3aed;letter-spacing:1px">CHALLENGE 5 · ML FLAVOR</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">P(class | feature)</div>
<div style="color:#4a5578;margin-top:6px">Of 500 reviews, 200 are negative. The word 'refund' appears in 120 negative reviews and 30 positive ones. Find P(negative | 'refund').</div>
</div>

In [6]:
neg_refund=120; pos_refund=30
p_neg_given_refund = neg_refund / (neg_refund + pos_refund)
print(f"reviews containing \"refund\": {neg_refund+pos_refund}")
print(f"P(negative | refund) = 120/150 = {p_neg_given_refund:.2f}")
print(f"compare P(negative) = 200/500 = {200/500:.2f}  -> the word \"refund\" is a strong negative signal")

reviews containing "refund": 150
P(negative | refund) = 120/150 = 0.80
compare P(negative) = 200/500 = 0.40  -> the word "refund" is a strong negative signal


**Answer:** of the 150 reviews containing "refund", 120 are negative, so P(negative | "refund") = 120/150 = 0.80, far above the base rate P(negative) = 0.40. A sentiment classifier is built from exactly these conditional probabilities: each feature shifts the estimated P(class | features) up or down.

---
<div style="background:#ffffff;border:1px solid #e6e9f2;border-radius:16px;padding:22px 26px;font-family:Inter,sans-serif">
<div style="font-size:19px;font-weight:800;color:#1a2138">🎉 Nicely done</div>
<div style="color:#4a5578;line-height:1.8;margin-top:8px">You read conditional probabilities off tables, multiplied through dependent draws, tested independence by conditioning, fell for (then escaped) the base-rate trap, and computed P(class | feature) the way a classifier does. Bayes' theorem, next, turns these conditionals around.</div>
</div>

<div style="text-align:center;color:#8b94b3;font-size:12px;margin-top:16px">Statistics, Data Science and AI: A Visual Handbook · © 2026 John Fisher</div>